test/conftest.py

```diff
@@ -2,12 +2,27 @@
 
 import pytest
 
+from clinvar_ingest import config
+
 
 @pytest.fixture(scope="session", autouse=True)
 def env_config():
+    """
+    Overrides clinvar_ingest.config values.
+    Also returns a SimpleNamespace with the same values for use in tests.
+
+    The main reason this is in conftest.py is that it gets run by pytest first,
+    which enables setting environment values used in the FastAPI app before
+    clinvar.api.main is imported, necessary since the app is constructed upon import
+    and checks whether the required environment variables are set.
```

Comment:
One suggestion for the config... why not just use a pydantic model with typed fields:
```
Config(BaseModel):
  bucket_name: str
  bucket_staging_prefix: str
  timeout_s: int
```

it's kinda the same as using a namespace... just a little more user friendly!

Reply:
Would you put the global config variables (in config.py) in a model like this, or are you just suggesting it for inside the test code?

Reply:
Originally I was putting it in a namespace in the fixture function and returning it, because I was using it as a `patch` object in individual test cases, but since the values in the config have to be set prior to importing `clinvar_ingest.api.main` (which can't be done if patching the values inside individual test functions), now I'm just setting the attributes directly on `clinvar_ingest.config`

Reply:
I'm doing [this](https://github.com/broadinstitute/seqr-loading-pipelines/blob/main/v03_pipeline/lib/model/environment.py#L21-L28) in the pipeline... which is sort of an approximation of what I was thinking because I don't have pydantic there.  So yeah, suggesting it outside of the test code and in general.  I've honestly found having a typed Config object super ergonomic for dealing with environment variables that are integers or enums.  You can do it with globals in a namespace, it's just more direct and pythonic.

I do agree that managing this without mocking is probably the right thing.  You can just have the `pytest` fixture set attributes on `Config` and be done with it.  With mocking you need to mock the object in each different file where the `Config` is accessed, which is super annoying.

Reply:
Made some changes to use a Pydantic class for validating env configs

https://github.com/clingen-data-model/clinvar-ingest/compare/969d2f189fd8b7a1b293bea045420b251c0da96e..325b9ae4d9d96eb45a17c97e4ca9c2de1cc8e414

I still don't like the way the test overriding over those values works, and the use of lots of globals

Reply:
Added issue here to address/discuss further changes

https://github.com/clingen-data-model/clinvar-ingest/issues/73